pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamDataDecoderImpl.java

```diff
@@ -34,6 +34,13 @@ public class StreamDataDecoderImpl implements StreamDataDecoder {
   private final StreamMessageDecoder _valueDecoder;
   private final GenericRow _reuse = new GenericRow();
 
+  /**
+   * @return Whether the given key is one of the special types of keys (__KEY, header, metadata, etc.)
+   */
+  public static boolean isSpecialKeyType(String key) {
```

Comment:
It's used so that we can *skip* transforming these fields in the `JsonLogTransformer`. I put it in this class so that if a developer changes the list of special fields in this class, they know to update this method as well.